provisioning/device/src/transport_state_machine.ts

```diff
@@ -45,12 +45,12 @@ export class TransportStateMachine extends EventEmitter implements Provisioning.
               callback(err, body, result);
             }
           },
-          connect: (callback) => {
-            this._fsm.transition('connecting', callback);
+          connect: (authorization, callback) => {
+            this._fsm.transition('connecting', callback, authorization);
```

Comment:
>callback, authorization [](start = 47, length = 23)

Odd parameter order.  callbacks usually last.

Reply:
yes. callback last please 

---
In reply to: [147507291](https://github.com/Azure/azure-iot-sdk-node/pull/125#discussion_r147507291) [](ancestors = 147507291)

Reply:
actually, I'm realizing callback comes first in most of the state machines, that take an error or a result. so now I'm torn. let's talk.